src/main/resources/blockly/js/mcreator_blockly.js

```diff
@@ -124,4 +124,16 @@ function workspaceToXML() {
     }
 
     return Blockly.Xml.domToText(treeXml);
+}
+
+class EntryChange extends Blockly.Events.BlockChange {

```

Comment:
An entry is not the same thing as a value so it needs a separate method. And obviously an event (using regular `BlockChange` instead would update field's value but not entry, probably resulting in no visual effect in the editor).

Reply:
✔️ Fixed

Reply:
The issue with this fix is that now mutators don't work anymore on data list selectors

Reply:
Data list fields that are added by the mutator?

Reply:
Forgot that one. But the class validation (not local validation you're talking about) is not the same as important here, right?

Reply:
It can help with #3503, as it allows changing the input types directly from the validator. The fact that the validator isn't executed means there's something wrong with how the field is defined

Reply:
That's about the local validation.

Reply:
I'll bring it back in the following minutes.